### PR TITLE
fixes a bug where the parampopup wouldn't refresh or remove

### PR DESCRIPTION
**Pull request by kasperpeulen**
not all clicks where catched by document.onClick

### CODE REVIEW

File: lib/editing/editor.dart [hunk 1]
```diff
@@ -60,6 +60,13 @@ abstract class Editor {
 
   bool get hasFocus;
 
+  /**
+   * Fired when a mouse is clicked. You can preventDefault the event to signal
+   * that the editor should do no further handling.  Only implemented for
+   * codemirror, returns `null` for ace editor and comid.
+   */
+  Stream get onMouseDown;
```
Comment (devoncarew) [inline lib/editing/editor.dart]: Can you parameterize this type? `Stream<MouseEvent>`, here, and in the subclasses.

File: lib/editing/editor_ace.dart [hunk 2]
```diff
@@ -139,6 +139,9 @@ class _AceEditor extends Editor {
   // TODO: Add a cursorCoords getter for ace.
   Point get cursorCoords => null;
 
+  // TODO: Add a onMouseDownn getter for ace.
```
Comment (devoncarew) [inline lib/editing/editor_ace.dart]: sp: onMouseDownn

File: lib/editing/editor_codemirror.dart [hunk 3]
```diff
@@ -187,6 +187,10 @@ class _CodeMirrorEditor extends Editor {
 
   bool get hasFocus => cm.jsProxy['state']['focused'];
 
+  Stream get onMouseDown {
```
Comment (devoncarew) [inline lib/editing/editor_codemirror.dart]: It's preferred to use the method shortcut for small methods:
```
Stream<MouseEvent> get onMouseDown => cm.onMouseDown;
```

**Comment by devoncarew** [comment]
Thanks! There are one or two smaller items we'll want to address before landing this.

File: lib/editing/editor.dart (hunk 1 above)
Comment (kasperpeulen) [inline lib/editing/editor.dart]: Webstorm tells me this if I try that:

`Warning:(68, 10) The name 'MouseEvent' is not a type and cannot be used as a parameterized type`

Comment (devoncarew) [inline lib/editing/editor.dart]: Oh dear. I'll merge and see if I see the same issue -

**Comment by devoncarew** [comment]
merging -

**Pull request merged**